lambda/MAS.Tests/IntergrationTests/MailControllerTests.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Threading.Tasks;
+using Xunit;
+using Shouldly;
+using Moq;
+using MAS.Tests.Infrastructure;
+using MAS.Services;
+using MAS.Models;
+using MAS.Controllers;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+
+namespace MAS.Tests.IntergrationTests
+{
+    public class MailControllerTests : TestBase
+    {
+        [Fact]
+        public async Task Put()
+        {
+            //Arrange
+            var mockContentService = new Mock<IContentService>();
+            mockContentService.Setup(x => x.GetItemAsync(It.IsAny<string>())).ReturnsAsync(JsonConvert.DeserializeObject<Item>("{ \"Id\" : \"1234\", \"Title\" : \"My Test Drug\" }"));
+
+            var mockMailService = new Mock<IMailService>();
+            mockMailService.Setup(x => x.CreateAndSendCampaignAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>()));
+
+            var mailController = new MailController(mockMailService.Object, mockContentService.Object);
+
+            //Act
+            var response = await mailController.PutAsync() as JsonResult;
+
+            // Assert
+            response.Value.ShouldBe("posted");
```

Comment:
Now this is interesting. We have to ask ourselves here what are we actually testing? After all it's the behaviour of the code we want to test to make sure it does what it's intended to do and if someone changes it we want our tests to pick up on that but that only happens if the tests are "tightly coupled" (not referring to the other tight coupling which is normally bad) to the behaviour. So for example lets take a look at: mailController.PutAsync()

What if someone removed 

  ```var subject = "MAS Email";
     var previewText = "This MAS email was created " + DateTime.Now.ToShortDateString();
     await _mailService.CreateAndSendCampaignAsync(subject, previewText, body);
```

The test would still pass! 😲😲😲😲😲😲😲

Because the test is only asserting the return which is:
`return new JsonResult("posted");`

So firstly I think the code for the return statement needs to check if it has infact been posted or not because currently it's telling porkies! 🐷🐷🐷🐷 :-)